### PR TITLE
feat: add in app guidance

### PR DESCRIPTION
**Pull request by msrivastav13**
### What does this PR do?

This PR adds  v1 version of in-app guidance to the easy spaces app

### What issues does this PR fix or reference?

#<Insert GitHub Issue>

## The PR fulfills these requirements:

[ ] Tests for the proposed changes have been added/updated.
[ ] Code linting and formatting was performed.

### Functionality Before

<insert gif and/or summary>

### Functionality After

<insert gif and/or summary>

### CODE REVIEW

File: config/project-scratch-def.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "orgName": "Easy Spaces",
     "edition": "Developer",
     "hasSampleData": false,
-    "features": ["ServiceCloud"],
+    "features": ["Walkthroughs"],
```
Comment (msrivastav13) [inline config/project-scratch-def.json:5]: @adityanaag3 FYI - I found that we do not need the ServiceCloud enabled for this app at all. Fixing it to just use the Walkthroughs feature.

File: README.md [hunk 2]
```diff
@@ -72,6 +72,14 @@ There are two ways to install Easy Spaces:
     sfdx force:user:permset:assign -n SpaceManagementApp
     ```
 
+1. (Optional) Assign the `Walkthroughs` permission set to the default user.
```
Comment (msrivastav13) [inline README.md:75]: This adds instructions to assign the necessary permission set. This is auto-created in orgs when the walkthroughs feature is enabled.

File: config/project-scratch-def.json [hunk 3]
```diff
@@ -17,6 +17,10 @@
         },
         "pathAssistantSettings": {
             "pathAssistantEnabled": true
+        },
+        "userEngagementSettings": {
```
Comment (msrivastav13) [inline config/project-scratch-def.json:21]: @adityanaag3 FYI - This was causing those walkthroughs to not appear.

File: es-space-mgmt/main/default/prompts/CreateReservationPrompt.prompt-meta.xml [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        <targetPageKey1>Reservation_Manager</targetPageKey1>
+        <targetPageType>standard__navItemPage</targetPageType>
+        <title>View Reservation Record</title>
+        <userAccess>Everyone</userAccess>
+        <userProfileAccess>Everyone</userProfileAccess>
+        <versionNumber>1</versionNumber>
+    </promptVersions>
+    <promptVersions>
+        <actionButtonLabel>Next Tour</actionButtonLabel>
+        <actionButtonLink>/lightning/n/Spaces_Designer</actionButtonLink>
```
Comment (msrivastav13) [inline es-space-mgmt/main/default/prompts/CreateReservationPrompt.prompt-meta.xml:124]: This will link both the code tours.

**Review by adityanaag3 — CHANGES_REQUESTED** [review]
Perfect. I really like the use of emojis to point to the relevant places. However, noticed 2 things
1. The very first prompt can be at the bottom center to get attention. Right now it's on the left overlapping with the already white background.
2. The way you linked tours is good but here is what is happening - The last prompt of the first tour is shown in the Space Designer tab (we talk about CMDT and Object agnostic design in that - which is actually a part of the Reservation Manager tab). Because of this, the second walkthrough automatically launches once I click on "Finish" in the first tour, (coz the second WT associated with the Space Designer tab) making the other button redundant. My preference would be a single WT that covers both the pages - that way the user sees the entire app flow instead of having to decide whether or not they want a tour of what to do next.

**Review by adityanaag3 — APPROVED** [review]

**Pull request merged**